caravel/assets/javascripts/explorev2/reducers/exploreReducer.js

```diff
@@ -2,30 +2,23 @@ import { defaultFormData, defaultOpts } from '../stores/store';
 import * as actions from '../actions/exploreActions';
 import { addToArr, removeFromArr, alterInArr } from '../../../utils/reducerUtils';
 
-const setFormInViz = function (state, action) {
-  const newFormData = Object.assign({}, state);
-  newFormData[action.key] = action.value;
+const setFormInViz = function (oldFd, action) {
```

Comment:
we also use this function for Select and TextField, for checkbox action.value is undefined while for other fields they are directly set to action.value